variables.tf

```
@@ -71,20 +88,64 @@ variable "pubsub_maximum_backoff" {
   default     = "600s"
 }
 
+variable "function_roles" {
+  description = <<-EOF
+    A list of IAM roles to give the Cloud Function.
+  EOF
+  type        = list(string)
+
+  default = [
+    "roles/compute.viewer",
+    "roles/iam.serviceAccountViewer",
+    "roles/cloudscheduler.viewer",
+    "roles/cloudasset.viewer",
+    "roles/browser", # for viewing projects
+  ]
+}
+
+variable "enable_function" {
+  description = "Whether to enable the Cloud function"
+  type        = bool
+  default     = true
+}
+
+variable "function_bucket" {
+  description = "GCS bucket containing the Cloud Function source code"
+  type        = string
+  default     = "observeinc"
+}
+
+variable "function_object" {
+  description = "GCS object key of the Cloud Function source code zip file"
+  type        = string
+  default     = "google-cloud-functions.zip"
+}
+
+variable "function_schedule" {
+  description = <<-EOF
+    How often to trigger the cloud function. This is a Cloud Scheduler Job schedule:
+    https://cloud.google.com/scheduler/docs/reference/rest/v1/projects.locations.jobs#Job
+  EOF
+  type        = string
+  default     = "*/5  * * * *"
+}
+
+# tflint-ignore: terraform_unused_declarations
 variable "project_id" {
-  description = "Project ID"
+  description = "Project ID (deprecated)"
   type        = string
+  default     = ""
```

```suggestion
  default     = null
```

Prefer `null` over zero values to represent "empty" variables that are unset. Terraform treats `my_attr = null` as equivalent to omitting `my_attr`. An empty string, list, etc is considered an explicitly set value.

---

done